document get_name() and get_version() (#1444)
conan-io/docs: reference/conanfile/methods.rst

.. _method_configure_config_options:

get_name(), get_version()
--------------------------

> Allow defining the name and version of the package dynamically. The following example defines the package name reading from

Reviewer: Suggestion for the text:

```
Dynamically define "name" and "version" attributes in the recipe with these methods. The following example
defines the package name reading it from a "name.txt" file and the version from the branch and commit of the
recipe's repository.
```

a *name.txt* text file, and the version from the branch and commit of the git repository in which this recipe lives.

..  code-block:: python

    from conans import ConanFile, tools

    class HelloConan(ConanFile):
        def get_name(self):
            self.name = tools.load("name.txt")

        def get_version(self):
            git = tools.Git()
            self.version = "%s_%s" % (git.get_branch(), git.get_revision())

The ``get_name()`` and ``get_version()`` methods should respectively set the ``self.name`` and ``self.version`` attributes.

> These methods are only executed when the recipe is in user space, that is, in the ``export`` and ``create`` commands, and the

Reviewer: ```suggestion
These methods are only executed when the recipe is in userspace (:command:`export` and :command:`create` commands) and the
```

Author: userspace? I don't think that exists

Reviewer: Not sure, it was suggested by grammarly

Reviewer 2: ![image](https://user-images.githubusercontent.com/3807515/67272277-90178400-f4bc-11e9-82ef-4bc5aba42915.png)

Author: Changed to user folder, for clarity

> ``install <path>`` when the path points to a *conanfile.py* containing these methods.

Reviewer: I don't understand this sentence. Maybe it needs some rephrasing

Reviewer 2: Agree. maybe "when calling `conan install <path>`"

> See more examples :ref:`in this howto <capture_version>`.

Reviewer: Add this in a ``.. seealso::`` box

.. _method_configure_config_options:

set_name(), set_version()
--------------------------

> Dynamically define "name" and "version" attributes in the recipe with these methods. The following example

Reviewer: ```suggestion
Dynamically define ``name`` and ``version`` attributes in the recipe with these methods. The following example
```

> defines the package name reading it from a "name.txt" file and the version from the branch and commit of the

Reviewer: ```suggestion
defines the package name reading it from a *name.txt* file and the version from the branch and commit of the
```
